bundles/specmate-cause-effect-patterns/src/com/specmate/cause_effect_patterns/resolve/XTextUtil.java

```
@@ -0,0 +1,69 @@
+import com.specmate.cause_effect_patterns.internal.specDSL.Model;
+import com.specmate.cause_effect_patterns.internal.specDSL.Rule;
+import com.specmate.cause_effect_patterns.parse.matcher.MatchRule;
+import com.specmate.cause_effect_patterns.parse.matcher.Matcher;
+import com.specmate.cause_effect_patterns.parse.matcher.MatcherException;
+import com.specmate.cause_effect_patterns.resolve.util.RuleUtility;
+
+public class XTextUtil {
+	public static List<MatchRule> generateMatchers(URI resourcePath, URI... definitionPaths) {
+		// do this only once per application
+		Injector injector = new SpecDSLStandaloneSetup().createInjectorAndDoEMFRegistration();
+		 
+		// obtain a resourceset from the injector
+		XtextResourceSet resourceSet = injector.getInstance(XtextResourceSet.class);
+		 
+		// load a resource by URI, in this case from the file system
+		Resource resource = resourceSet.getResource(resourcePath, true);
+		//Add any other models here
+		for(URI res: definitionPaths) {
+			resourceSet.getResource(res, true);
+		}
+		
+		// Validation
+		IResourceValidator validator = ((XtextResource)resource).getResourceServiceProvider().getResourceValidator();
+		List<Issue> issues = validator.validate(resource, CheckMode.ALL, CancelIndicator.NullImpl);
+		for (Issue issue : issues) {
+		  System.err.println(issue.getMessage());
+		}
+		if(issues.size() > 0) {
+			return null;
+		}
+		
+		List<MatchRule> out = resource.getContents().stream()
+				.filter(e -> e instanceof Model)
+				.flatMap(m -> ((Model)m).getElements().stream())
+				.filter(e -> e instanceof Rule)
+				.map(r -> resolveRule((Rule)r))
+				.filter(e -> e != null)
+				.collect(Collectors.toList());
+		
+		return out;
+	}
+	
+	private static MatchRule resolveRule(Rule r) {
+		try {
+			Matcher match = RuleUtility.transform(r);
+			return new MatchRule(match, r.getName());	
+		} catch (MatcherException e) {
+			e.printStackTrace();
```

I would rather throw this exception and let the client handle it